### PR TITLE
[CTRI-35] Test HTTP Client for CTR Python API Module

### PR DESCRIPTION
**Pull request by gdavoian**

### CODE REVIEW

File: tests/test_api/test_enrich/test_enrich.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+from threatresponse.api.enrich import (
+    EnrichAPI,
+    DeliberateAPI,
+    ObserveAPI,
+    ReferAPI,
+)
+
+
+def test_composite_structure():
```
Comment (konstantin-ogulchansky) [inline tests/test_api/test_enrich/test_enrich.py]: This name does not actually tell anything about the subject being tested. I would propose to rename it to something like `test_types_of_inner_apis`.

Comment (gdavoian) [reply tests/test_api/test_enrich/test_enrich.py]: Yes, makes sense. Will rename.

File: tests/test_client.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    # Verify that only a single request has been made to an external API.
+    # Don't check the actual arguments since we're not interested in any
+    # auth-specific details here.
+    inner_session_request.assert_called_once()
+
+    logger.info.assert_called_once_with(
+        'POST https://visibility.amp.cisco.com/iroh/oauth2/token 200 OK'
+    )
+
+    assert isinstance(client.inspect, InspectAPI)
```
Comment (konstantin-ogulchansky) [inline tests/test_client.py]: These checks should be moved to another test.

File: tests/test_client.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    mocked.ok = 100 <= status_code < 400
+
+    if mocked.ok:
+        mocked.json.return_value = {'access_token': 'ACCESS_TOKEN'}
+
+    else:
+        mocked.json.return_value = {'error': 'ERROR'}
+
+        error = HTTPError('Some error message here.', response=mocked)
+        mocked.raise_for_status.side_effect = error
```
Comment (konstantin-ogulchansky) [inline tests/test_client.py]: Let's restructure this code like this:
``` python
error = HTTPError('Some error message here.', response=mocked)

mocked.json.return_value = {'error': 'ERROR'}
mocked.raise_for_status.side_effect = error
```

Comment (gdavoian) [reply tests/test_client.py]: Done

File: tests/test_client.py (hunk 2 above)
Comment (gdavoian) [reply tests/test_client.py]: Done

**Pull request merged**